assertions.js

```diff
@@ -22,20 +22,35 @@ export const shouldBeBs58 = s => bs58.test(s);
 
 export const shouldBeBase64NoPadUrl = s => BASE_64URL_NOPAD_REGEX.test(s);
 
+/**
+ * Posts a VC to a verifier and asserts on the response.
+ *
+ * @param {object} options - Options to use.
+ * @param {object} options.credential - The credential to post.
+ * @param {object} options.verifier - The verifier to post to.
+ * @param {string} [options.reason] - The expected reason for the failure.
+ * @param {object} [options.options] - Request specific options.
+ *
+ * @returns {Promise<object>} The result of the verification.
+ */
 export const verificationFail = async ({
   credential, verifier, reason, options = {}
 } = {}) => {
+  const {settings: {options: verifierOptions}} = verifier;
   const body = {
     verifiableCredential: credential,
     options: {
-      checks: ['proof'],
+      ...verifierOptions,
+      // request specific options should over endpoint options
       ...options
     }
   };
   const {result, error} = await verifier.post({json: body});
-  should.not.exist(result, 'Expected no result from verifier.');
-  should.exist(error, 'Expected verifier to error.');
+  const withReason = reason || '';
+  should.not.exist(result, 'Expected no result from verifier.' + withReason);
+  should.exist(error, 'Expected verifier to error.' + withReason);
```

Comment:
We should avoid these `.exist` style tests per https://www.chaijs.com/api/bdd/#exist

And we should use chaining rather than passing stuff in via method calls (which isn't how `.exist` typically works). The following would be more explicit:
```js
result.should.equal('', 'Expected no result from verifier.' + withReason);
```

Reply:
if result is undefined that will throw. I think `expect` is better than should.